src/options.ts

```diff
@@ -1,9 +1,28 @@
-function getInputValue(elementId: string): string {
-  return (document.getElementById(elementId) as HTMLInputElement).value;
+function getInputValue(elementId: string): string | boolean {
+  const elm = document.getElementById(elementId)! as HTMLInputElement;
+  if (elm) {
+    if (elm.type === "checkbox") {
+      return elm.checked;
+    } else {
+      if (elm.value === "true") return true;
+      if (elm.value === "false") return false;
+
+      return elm.value;
+    }
+  }
+  throw new Error('Invalid input element');
 }
 
-function setInputValue(elementId: string, value: string): void {
-  (document.getElementById(elementId) as HTMLInputElement).value = value;
+function setInputValue(elementId: string, value: string | number | boolean): void {
+  const elm = document.getElementById(elementId)! as HTMLInputElement;
+  if (elm) {
+    if (elm.type === "checkbox") {
+      elm.checked = !!value;
```

Comment:
I am curious, why the use of a double `!`?

Reply:
Is it alias for `Boolean(value)`, this is using, if you want change data type to bool.